### ISSUE TITLE
bug(assembly-loading): creating a moq of an interface in a third-party assembly can sometimes throw a FileNotFoundException

### ISSUE THREAD

**Issue by dmccaffery** [moq/moq4#299]
I created a simple reproduction in a repo: https://github.com/dmccaffery/moq-repro

execute reproduce.sh or reproduce.ps1 to use an isolated copy of dotnet core 1.1, or simply do:

```cmd
dotnet restore
dotnet test
```

in each project under the test folder. One will work, one will not. The only difference between the two is a different version of the same dependency. The one that fails is actually the current production version of the dependency.

**Comment by stakx** [comment]
I can't get your `reproduce.ps1` to run without errors, so I just looked at the actual test code and ran below program without the following NuGet package dependencies. It ran without throwing any exceptions.

* `Microsoft.NETCore.App` 1.1.2
* `Microsoft.Build.Framework` 15.1.1012
* `Moq` 4.7.25

```csharp
using Microsoft.Build.Framework;
using Moq;

class Program
{
    static void Main(string[] args)
    {
        Mock.Of<IBuildEngine>();
        Mock.Of<IBuildEngine4>();
    }
}
```

Could you please update your repro code to current versions of .NET Core, Microsoft.Build.Framework, and Moq, and see if the error persists?

**Comment by stakx** [comment]
Having a look at this again. I believe the error you encountered might be due to the way how `Microsoft.Build.Framework` version 14.3.0 is made up.

[According to the package page on NuGet.org](https://www.nuget.org/packages/Microsoft.Build.Framework/14.3.0), this package does target .NET Standard 1.3 (since the page lists dependencies for that target). If you download and extract the package, you will see the following contents:

    lib\net45\Microsoft.Build.Framework.dll
    ref\net45\Microsoft.Build.Framework.dll
    ref\netstandard1.3\Microsoft.Build.Framework.dll
    Microsoft.Build.Framework.nuspec

Note the absence of a `lib\netstandard1.3` folder with the actual assembly. (This would be the relevant one, as `net45` should be unusable on the .NET Core platform.) The assembly under `ref\netstandard1.3` appears to be a reference assembly; it does not contain any working implementations, and it will not get copied to your program's output directory. Therefore, I would assume that if you do not have version 14.3.0 of `Microsoft.Build.Framework` installed system-wide, you're going to get the `FileNotFoundException`.

Contrast this with `Microsoft.Build.Framework` version 15.1.262-preview5. This package contains the following:

    lib\net46\Microsoft.Build.Framework.dll
    lib\net46\Microsoft.Build.Framework.xml
    lib\netstandard1.3\Microsoft.Build.Framework.dll
    lib\netstandard1.3\Microsoft.Build.Framework.xml
    Microsoft.Build.Framework.nuspec

So in conclusion, I would say that this is not an error with Moq, but with the way how that Microsoft NuGet package was made available.
